internal/pkg/cli/task_delete.go

```diff
@@ -262,26 +290,131 @@ func (o *deleteTaskOpts) askTaskName() error {
 		if err != nil {
 			return fmt.Errorf("select task from default cluster: %w", err)
 		}
-		o.name = task.Name
+		o.name = task
 		return nil
 	}
 	task, err := sel.Task(taskDeleteNamePrompt, "", selector.TaskWithAppEnv(o.app, o.env))
 	if err != nil {
 		return fmt.Errorf("select task from environment: %w", err)
 	}
-	o.name = task.Name
+	o.name = task
 	return nil
 }
 
 func (o *deleteTaskOpts) Execute() error {
-	// Get clients.
-
-	// Stop tasks.
+	// Stop all running tasks in given family.
+	if err := o.stopTasks(); err != nil {
+		return err
+	}
 
 	// Clear repository.
+	if err := o.clearECRRepository(); err != nil {
+		return err
+	}
 
 	// Delete stack.
+	if err := o.deleteStack(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *deleteTaskOpts) stopTasks() error {
+	sess, err := o.getSession()
+	if err != nil {
+		return fmt.Errorf("get session: %w", err)
+	}
+
+	o.spinner.Start(fmt.Sprintf("Stopping all running tasks in family %s.", color.HighlightUserInput(o.name)))
+
+	// Stop tasks.
+	if o.defaultCluster {
+		if err = o.newTaskStopper(sess).StopDefaultClusterTasks(o.name); err != nil {
+			o.spinner.Stop(log.Serrorln("Error stopping running tasks in default cluster."))
+			return fmt.Errorf("stop running tasks in family %s: %w", o.name, err)
+		}
+	} else {
+		if err = o.newTaskStopper(sess).StopAppEnvOneOffTasks(o.app, o.env, o.name); err != nil {
+			o.spinner.Stop(log.Serrorln("Error stopping running tasks in environment."))
+			return fmt.Errorf("stop running tasks in family %s: %w", o.name, err)
+		}
+	}
+	o.spinner.Stop(log.Ssuccessf("Stopped all running tasks in family %s.\n", color.HighlightUserInput(o.name)))
+	return nil
+}
+
+func (o *deleteTaskOpts) clearECRRepository() error {
+	// ECR Deletion happens from the default profile in app delete. We can do it here too by getting
+	// a default session in whichever region we're deleting from.
+	var defaultSess *awssession.Session
+	var err error
+	if o.defaultCluster {
+		defaultSess, err = o.sess.Default()
+		if err != nil {
+			return fmt.Errorf("get default session for ECR deletion: %s", err)
+		}
+	} else {
+		regionalSession, err := o.getSession()
+		if err != nil {
+			return err
+		}
+		defaultSess, err = o.sess.DefaultWithRegion(aws.StringValue(regionalSession.Config.Region))
+		if err != nil {
+			return fmt.Errorf("get default session for ECR deletion: %s", err)
+		}
+	}
+	info, err := o.getTaskInfo()
+	if err != nil {
+		return err
+	}
+	o.spinner.Start(fmt.Sprintf("Emptying ECR repository for task %s.", color.HighlightUserInput(o.name)))
+	err = o.newImageRemover(defaultSess).ClearRepository(info.ECRRepoName())
```

Comment:
I was wondering if we need `GetTaskStack`: looks like the reason why we get the task stack is to get the ECR repo name, which is simply `"copilot"+o.name`. Can we make best efforts trying to empty the ECR repo?

Also I think we should try to make sure `copilot task delete` is idempotent. Users shouldn't get interrupted if any of the resources have already been deleted before running the command. What do you think?

Reply:
We keep the TaskStackInfo around because that struct is the input to `DeleteTask` and it's convenient to use to grab the ECR Repo name. Since we cache it we only have to call the underlying method once.

Reply:
Oh got it. Saw how the info gets cached. But looks like it will error out if the stack doesn't exist? Maybe we should format the ECR repo name here for repo cleaning, and do `GetTaskStack` when we are trying to delete the stack. And skip deleting if the stack doesn't exist. What do you think?

Reply:
Okay, I've added some idempotency checking for the error which gets returned when the stack doesn't exist and added a unit test for that case.